theia-extensions/viewer-prototype/src/browser/trace-explorer/trace-explorer-sub-widgets/charts-cheatsheet-component.tsx

```diff
@@ -0,0 +1,215 @@
+import * as React from 'react';
+import { inject, injectable, postConstruct } from 'inversify';
+import { Dialog, DialogProps } from '@theia/core/lib/browser/dialogs';
+import { ReactDialog } from '@theia/core/lib/browser/dialogs/react-dialog';
+import { Message } from '@theia/core/lib/browser/widgets/widget';
+
+@injectable()
+export class ChartShortcutsDialogProps extends DialogProps {
+}
+
+@injectable()
+export class ChartShortcutsDialog extends ReactDialog<void> {
+
+    constructor(
+        @inject(ChartShortcutsDialogProps) protected readonly props: ChartShortcutsDialogProps
+    ) {
+        super({
+            title: 'Trace Viewer Keyboard and Mouse Shortcuts',
+        });
+        this.appendAcceptButton(Dialog.OK);
+    }
+
+    @postConstruct()
+    protected async init(): Promise<void> {
+        this.title.label = 'Trace Viewer Keyboard and Mouse Shortcuts';
+        this.update();
+    }
+
+    protected render(): React.ReactNode {
```

Comment:
Looking at the implementation it's quite hard-coded and there is code duplication. Maybe be could have a sub-components that you supply to different sections which then easily can be modified when short-cuts are modified or added. I'm ok if it's done in a separate PR as a separate task, because this is a good UX improvement and it's good to bring it to the users quickly. The follow-up PR should be done soon though.

Reply:
> Looking at the implementation it's quite hard-coded and there is code duplication. Maybe be could have a sub-components that you supply to different sections which then easily can be modified when short-cuts are modified or added. I'm ok if it's done in a separate PR as a separate task, because this is a good UX improvement and it's good to bring it to the users quickly. The follow-up PR should be done soon though.

@bhufmann Refactored the code w/ sub-components for each section and created a separate section for time-graph specific actions:
![Screen Shot 2022-01-25 at 11 21 57 AM](https://user-images.githubusercontent.com/92893187/151027093-d0052e8e-dd67-4eca-80a0-64cf9841f981.png)